### PR TITLE
fix: improve compatibility of Google Cloud Functions, `req.body` is the parsed JSON payload

### LINKED ISSUE #935: [BUG]: google-cloud-function, can not verify payload
### What happened?

The request.body is already provided by gcf as an object. We expect that Lambda can provide request.body as a string but as it is an object in gcf it can not verify the request.body, because to verify the request.body is just using toString() of the object but not JSON.stringify.

### Versions

v. 12.0.8

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Uzlopak**
Resolves #935 

As requested the light solution.

@wolfy1339 

Resolves #ISSUE_NUMBER

----

### Before the change?

* 

### After the change?

* 

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No

----

### CODE REVIEW

File: src/middleware/node/middleware.ts [hunk 1]
```diff
@@ -93,7 +93,24 @@ export async function middleware(
   }, 9000).unref();
 
   try {
-    const payload = await getPayload(request);
```
Comment (G-Rath) [inline src/middleware/node/middleware.ts]: why are we not doing this in `getPayload`? it's purpose is literally to encapsulate this kind of logic

Comment (Uzlopak) [reply src/middleware/node/middleware.ts]: done.

**Comment by Uzlopak** [comment]
@G-Rath PTAL

**Comment by Uzlopak** [comment]
I go to sleep now. You can modify the code as you need. ;)

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**